fenix/app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -877,7 +858,7 @@ class HomeFragment : Fragment() {
             requireComponents.fenixOnboarding.finish()
             requireContext().components.appStore.dispatch(
                 AppAction.ModeChange(
-                    mode = currentMode.getCurrentMode(),
+                    mode = Mode.fromBrowsingMode(browsingModeManager.mode),
```

Comment:
so this essentially leaves onboarding and goes to the browsing mode?

Reply:
Since the onboarding will be separated from the home screen in this PR, the core idea is that we don't need to do any of the if/else checks on whether or not this needs to be handling onboarding logic anymore.

In https://github.com/mozilla-mobile/firefox-android/pull/1484/commits/a03c44a43f36a3674f1a251b6d0107f98acc298c#diff-b1935945f0af44dcf0d6f2f46ee58afbf72911042a1c487dc3074c78437091deL46, you will see that all we care about when we are not in onboarding is determining whether or not this is normal or private mode.

Reply:
Part 1 attempts to split up the logic around determining what mode the home screen should be in.